some style changes and fixes

UltimatChamp opened the pull request:
- Full-sized website window for a modern look.
- Added transitions to everything.
- A new look for badges.
  - Also, added a new Legacy badge.
- Provided a Hugo executable file like `gradlew` (fixes https://github.com/neoforged/websites/issues/33).
- Fixed the single-lined code block on notices to look unreadable in light mode.
- Covered some more IDEs in `.gitignore`.
- Fixes #36
- Check newer comments for more.

------------------
Preview URL: https://pr-34.neoforged-website-previews.pages.dev

sciwhiz12 commented on hugo.exe:
I don't think it is right for us to ship the Hugo binary in our repository. We should instead focus on fixing the underlying issue, which would also allow us to update to a newer Hugo version.

For now, please remove this file.

sciwhiz12 commented on themes/mainroad/assets/css/style.css:
```diff
@@ -1375,6 +1363,10 @@ summary:hover {
 	--color-highlight: #d7742f;
 	--color-badges-beta: #d7742f;
 	--color-badges-new: #e57;
+	--color-badges-legacy: #13cace;
```
The contrast between the background of the badge and the text color in dark mode (white) is bad.

![image](https://github.com/neoforged/websites/assets/21304337/e99ec184-5aa6-48de-89b3-e21a20ef8144)

UltimatChamp replied:
![image](https://github.com/neoforged/websites/assets/140878474/c193ea6c-84a5-4844-b315-32c2ca400edc)
I've changed the legacy badge's color in dark mode. _(btw i think these version numbers would look good)_

UltimatChamp replied:
![image](https://github.com/neoforged/websites/assets/140878474/edb6a517-f660-4bd1-8cb4-4c4ccbe3ac19)
this does not look good here...

sciwhiz12 commented on themes/mainroad/assets/css/style.css:
```diff
@@ -9,10 +9,7 @@
-
-html, body, .container--outer {
-	height: 100%;
+	transition: all 0.5s;
```
Why?

sciwhiz12 replied:
Huh. Neat!

sciwhiz12 commented on themes/mainroad/assets/css/style.css:
```diff
@@ -1360,7 +1348,7 @@ summary:hover {
 	--color-code-background: #bdc7c7;
 	--color-code-notice: #e57;
-	--color-code-notice-background: #bdc7c7;
+	--color-code-notice-background: #fff;
```
I assume this is the changed mentioned by:

> - Fixed the single-lined code block on notices to look unreadable in light mode.

Could you please point to an example of what this changes (preferrably, through a screenshot)?

UltimatChamp replied:
Before:
![image](https://github.com/neoforged/websites/assets/140878474/b788fd3a-679a-4e42-b226-770095846960)

After:
![image](https://github.com/neoforged/websites/assets/140878474/18af3fe8-cfe2-48b5-aa52-3bf697758adb)

Zoomed in, here, the code block may seem to be fine, but normally it's a little unreadable.

In all other places, the code blocks are unchanged.

sciwhiz12 replied:
Perhaps reduce the brightness of the new background by a small bit? So it's not _pure_ white. Otherwise, sure.

Though, should we also change the background of all other code blocks, for consistency? 🤔

sciwhiz12 reviewed (changes requested):
After consulting with others, it seems the old screen, limited in width it may be, looks better on wide screens rather than the changes in this PR. 

While there is a lot of wasted space on the existing layout, it's much more readable as the text is in a confined column, rather than stretching across the screen without balancing text size and other factors.

In line with that, please revert back to the fixed-width layout. We can discuss changing to a full-width layout in another PR, hopefully in a way that looks readable on wide screens.

UltimatChamp commented:
Hmm... I'll revert the changes you asked in a bit. It'll be nice if someone could make a better fullscreen solution; the current one is just pretty bugging.

UltimatChamp commented:
Ok!

Also, I'll record some visual representations of the changes you and Sci have asked for better comparison. _(and btw the highlight and the versions dropdowns were made pr-ed by me; just reading project-talk on discord lol)_

the transition effect is already on `neoforged.net`

UltimatChamp commented:
Does this look nice? I've removed the highlight on the dropdown text; it's now only on the badges.

https://github.com/neoforged/websites/assets/140878474/fbf76162-64ea-4ede-b3aa-baa7553edb4f

UltimatChamp commented:
- I've fixed the latest hugo crash, by disabling the `google analytics` feature.
- https://github.com/neoforged/websites/issues/36 has been fixed by renaming ~~Latest~~ News to `News`. _(This is already performed by the `Recent Posts` widget by showing the latest 5 news)_
- I've reverted back the fullscreen change.
- Hovering on the installer, the dropdown title will no longer be dimmed.
- The hovering feature is now only on the badges and the new version block on the dropwdown title.
- I've fixed the odd-looking toggle button, when it changes from a square to a circle. It's now from a squicircle to a circle.

UltimatChamp commented:
Should I do this?

https://github.com/neoforged/websites/assets/140878474/4af0adbb-13dc-4eb9-99a2-de7953f745f2

sciwhiz12 commented:
> Should I do this?
> rec_neo2.mp4

I think that's too distracting, personally.

sciwhiz12 commented on themes/mainroad/assets/css/style.css:
```diff
@@ -239,6 +248,11 @@ code {
 
+code:hover {
+	filter: brightness(.85);
+	cursor: pointer;
```
Why? It's a bit distracting to hover over a code block in a page and see it change to the hand pointer (for links and buttons), when it is not a button or a link. (It might also give the impression that you can click to copy?)

Also, is dimming the code block necessary? I think it should just be treated as ordinary text.

UltimatChamp replied:
> you can click to copy?

we can do that, actually.

OK, no more experimenting. I'll finalise the pr shortly.

sciwhiz12 reviewed (changes requested):
Thanks for figuring out the Hugo issue! We'll look into updating to the latest Hugo version on Cloudflare and here after this PR is merged.

The change in font size of the badges and code block in the version dropdowns are a bit too distracting, I think. At most, we could have them dim on hover, but changing size is a bit much. (And no other code block should really be dimming on hover, I think, to avoid association with e.g., buttons.)

sciwhiz12 commented on themes/mainroad/assets/css/style.css:
```diff
@@ -1076,13 +1071,11 @@ details[open] summary {
 
 summary code:hover {
-	font-size: 15px;
+        filter: brightness(.85);
```
I think you accidentally'd the indentation. 😄

sciwhiz12 reviewed (changes requested):
Looks good to me, just a small nitpick.

sciwhiz12 reviewed (approved):
Thanks!
